src/synthesize.jl

```diff
@@ -0,0 +1,195 @@
+    for elem in keys(abundances)
+        abundances[elem] /= total
+    end
+    abundances
+end
+
+"""
+Return a `Dict` which maps each species (i.e. Ba II) to number density [cm^-3]
+- `nₜ` the number density of atoms and mollecules in cm^-3
+- `nₑ` the number density of electrons in cm^-3
+- `temp` the termperature in K
+- `abundances` should be a `Dict` mapping species to absolute abundances between 0 and 1.
+"""
+function per_species_number_density(nₜ::Flt, nₑ::Flt, temp::Flt, abundances::Dict
+                                   ) where Flt <: AbstractFloat
+    #calculate the number density of each species
+    n_densities = Dict()
+    for (elem, abund) in abundances
+        if elem == "H"
+            #I'm not very happy with this, but I'm not sure what a better way to handle H is
+            weights = saha(ionization_energies["H"], [partition_funcs["H_I"], 
+                                                      partition_funcs["H_II"]],temp, nₑ)
+            n_densities["H_I"] = nₜ * abund * weights[1]
+            n_densities["H_II"] = nₜ * abund * weights[2]
+        else
+            weights = saha(ionization_energies[elem], 
+                           [partition_funcs[elem * "_I"], partition_funcs[elem * "_II"], 
+                              partition_funcs[elem * "_III"]], 
+                           temp, nₑ)
+            n_densities[elem * "_I"]   = nₜ * abund * weights[1]
+            n_densities[elem * "_II"]  = nₜ * abund * weights[2]
+            n_densities[elem * "_III"] = nₜ * abund * weights[3]
+        end
+    end
+    n_densities
+end
+
+"""
+The total continuum opacity, κ, at many frequencies, ν.
+
+- `νs` are frequencies in Hz
+- `T` is temperature in K
+- `nₑ` is the electron number density in cm^-3
+- `ρ` is the density in g cm^-3 
+- `number_densities` is a `Dict` mapping each species to its number density
+- `partition_funcs` is a `Dict mapping each species to its partition function
+"""
+function total_continuum_opacity(νs::Vector{F}, T::F, nₑ::F, ρ::F, number_densities::Dict, 
+                                 partition_funcs::Dict) where F <: AbstractFloat
+    κ = zeros(F, length(νs))
+
+    #Hydrogen continuum opacities
+    nH_I = number_densities["H_I"]
+    nH_I_div_U = nH_I / partition_funcs["H_I"](T)
+    κ += ContinuumOpacity.H_I_bf.(nH_I_div_U, νs, ρ, T) 
+    κ += ContinuumOpacity.H_I_ff.(nH_I, nₑ, νs, ρ, T)
+    κ += ContinuumOpacity.Hminus_bf.(nH_I_div_U, nₑ, νs, ρ, T)
+    κ += ContinuumOpacity.Hminus_ff.(nH_I_div_U, nₑ, νs, ρ, T)
+    #TODO uncomment when fixed
+    #κ += ContinuumOpacity.H2plus_bf_and_ff.(nH_I_div_U, number_densities["H_II"], νs, ρ, T)
+    
+    #He continuum opacities
+    nHe_II = number_densities["He_II"]
+    nHe_II_div_U = nHe_II / partition_funcs["He_II"](T)
+    κ += ContinuumOpacity.He_II_bf.(nHe_II_div_U, νs, ρ, T)
+    #TODO fix
+    #κ += ContinuumOpacity.He_II_ff.(nHe_II, nₑ, νs, ρ, T)
```

Comment:
```suggestion
    #κ += ContinuumOpacity.He_II_ff.(number_densities["He_III"], nₑ, νs, ρ, T)
```
Out of curiosity why is this commented but `ContinuumOpacity.He_II_bf` is not commented?

Reply:
Done.